src/input/input.style.tsx

```diff
@@ -22,8 +23,12 @@ export const InputElement = styled(BasicInput)<InputStyleProps>`
     ${(props) =>
         props.$styleType !== "no-border" &&
         css`
-            padding-left: ${Spacing["spacing-16"]};
-            padding-right: ${props.$showClear ? 0 : Spacing["spacing-16"]};
+            padding: ${Spacing["spacing-16"]}
+                ${props.$visuallyReadOnly
+                    ? 0
+                    : props.$showClear
+                    ? Spacing["spacing-16"]
+                    : Spacing["spacing-16"]};
```

Comment:
the logic should be:
padding left: if readonly, 0, else spacing-16
padding right: if readonly or showClear, 0, else spacing-16

this is because the clear button comes with its own padding already, so the additional right padding is not needed